server/src/main/java/org/elasticsearch/index/codec/postings/ES812PostingsReader.java

```diff
@@ -287,8 +285,6 @@ final class BlockDocsEnum extends PostingsEnum {
         private ES812SkipReader skipper;
         private boolean skipped;
 
-        final IndexInput startDocIn;
```

Comment:
The input is a final field on the outer class, no need to keep a separate reference to that index input in a non-static inner class.